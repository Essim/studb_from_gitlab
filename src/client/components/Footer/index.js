import React,{ Component } from "react";

export class Footer extends Component{
    constructor(props) {
        super(props);
        this.state = {
            window: undefined,
        }
    }

    componentDidMount() {
        this.setState({ window: window })
    }

    render(){
        if(this.state.window !== undefined)
            return (
                <footer className="main-footer">
                    <div className="pull-right hidden-xs">
                        Lien vers l'<b><a href={`${this.state.window.location.origin}/api-docs/`}>API</a></b>
                    </div>
                    <strong>Copyright &copy; 2017-2018 <a href="https://gitlab.com/shu-iro/studb">StudentDB</a>.</strong> All rights
                    reserved.
                </footer>
            );
        return(<footer className="main-footer">
            <div className="pull-right hidden-xs">
                Lien vers l'<b><a href={'/'}>API</a></b>
            </div>
            <strong>Copyright &copy; 2017-2018 <a href="https://gitlab.com/shu-iro/studb">StudentDB</a>.</strong> All rights
            reserved.
        </footer>)
    }
}

export default Footer;
