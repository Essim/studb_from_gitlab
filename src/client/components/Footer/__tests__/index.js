import React from 'react';
import Footer from '../';

test('render footer', ()=>{
    const wrapper = shallow(<Footer/>);
    expect(wrapper).toMatchSnapshot();
});