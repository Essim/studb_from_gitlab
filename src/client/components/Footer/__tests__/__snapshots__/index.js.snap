// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render footer 1`] = `
<footer
  className="main-footer"
>
  <div
    className="pull-right hidden-xs"
  >
    Lien vers l'
    <b>
      <a
        href="null/api-docs/"
      >
        API
      </a>
    </b>
  </div>
  <strong>
    Copyright Â© 2017-2018 
    <a
      href="https://gitlab.com/shu-iro/studb"
    >
      StudentDB
    </a>
    .
  </strong>
   All rights reserved.
</footer>
`;
